candle-wasm-examples/phi/phiWorker.js

```diff
@@ -0,0 +1,131 @@
+import init, { Model } from "./build/m.js";
+
+async function fetchArrayBuffer(url) {
+  const cacheName = "phi-mixformer-candle-cache";
+  const cache = await caches.open(cacheName);
+  const cachedResponse = await cache.match(url);
+  if (cachedResponse) {
+    const data = await cachedResponse.arrayBuffer();
+    return new Uint8Array(data);
+  }
+  const res = await fetch(url, { cache: "force-cache" });
+  cache.put(url, res.clone());
+  return new Uint8Array(await res.arrayBuffer());
+}
+class Phi {
+  static instance = {};
+
+  static async getInstance(weightsURL, modelID, tokenizerURL, quantized) {
+    // load individual modelID only once
+    if (!this.instance[modelID]) {
+      await init();
+
+      self.postMessage({ status: "loading", message: "Loading Model" });
+
+      const [weightsArrayU8, tokenizerArrayU8] = await Promise.all([
+        fetchArrayBuffer(weightsURL),
+        fetchArrayBuffer(tokenizerURL),
+      ]);
+
+      this.instance[modelID] = new Model(
+        weightsArrayU8,
+        tokenizerArrayU8,
+        quantized
+      );
+    }
+    return this.instance[modelID];
+  }
+}
+
+let controller = null;
+self.addEventListener("message", (event) => {
+  if (event.data.command === "start") {
+    controller = new AbortController();
+    generate(event.data);
+  } else if (event.data.command === "abort") {
+    controller.abort();
+  }
+});
+
+async function generate(data) {
+  const {
+    weightsURL,
+    modelID,
+    tokenizerURL,
+    quantized,
+    prompt,
+    temp,
+    top_p,
+    repeatPenalty,
+    seed,
+    maxSeqLen,
+  } = data;
+  try {
+    self.postMessage({ status: "loading", message: "Starting Phi" });
+    const model = await Phi.getInstance(
+      weightsURL,
+      modelID,
+      tokenizerURL,
+      quantized
+    );
+
+    self.postMessage({ status: "loading", message: "Initializing model" });
+    model.init_with_prompt(
```

Comment:
I think this return the first generated token but somehow it gets dropped at the moment?

Reply:
great catch! I completely missed that it return the first token 🤦🏼 need to fix the worker! thanks